### ISSUE TITLE
Failed to parse PKCS#12 file error:23076072:PKCS12 routines:PKCS12_parse:parse error

### ISSUE THREAD

**Issue by csubbiah** [edenhill/librdkafka#3276]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Thanks for this great library for .net consumers for kafka
We've 2 questions:
a. Require support on the PKCS12 Error when connecting to bootstrap with ssl using pkcs12
b. Pls share recommended nuget package to use for SchemaRegistry and AVRO serializer, deserializer which uses .net 4.5.2 framework only.

Description
===========
%3|1614176059.253|SSL|rdkafka#consumer-1| [thrd:app]: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt
%3|1614176059.254|SSL|rdkafka#consumer-1| [thrd:app]: error:23077074:PKCS12 routines:PKCS12_pbe_crypt:pkcs12 cipherfinal error
%3|1614176059.254|SSL|rdkafka#consumer-1| [thrd:app]: error:2306A075:PKCS12 routines:PKCS12_item_decrypt_d2i:pkcs12 pbe crypt error

How to reproduce
================
Good Day. We're a consumer and received our pkcs12 file shared by the kafka team to connect to kafka bootstrapserver. 
We're using Confluent kafka 1.5.3 targeting .net framework 4.5.2 
Here is the way we use configuration :

 BootstrapServers = "XXXXX",
 SecurityProtocol = SecurityProtocol.Ssl,
 SslKeystoreLocation = DirectoryName + "\\XXX.pkcs12",
 SslKeystorePassword = "XXXX",
 EnableSslCertificateVerification = false,
 SslEndpointIdentificationAlgorithm = SslEndpointIdentificationAlgorithm.None

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<REPLACE with e.g., v0.10.5 or a git sha. NOT "latest" or "current">`
 - [ ] Apache Kafka version: `<REPLACE with e.g., 0.10.2.3>`
 - [ ] librdkafka client configuration: `<REPLACE with e.g., message.timeout.ms=123, auto.reset.offset=earliest, ..>`
 - [ ] Operating system: `<REPLACE with e.g., Centos 5 (x64)>`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
Could be that the PKCS#12 store is created with a newer version of OpenSSL.
Check out this SO answer:
https://stackoverflow.com/a/43847627/1821055


Also, you really shouldn't disable SslCertificateVerification or you'll void the benefits of using SSL since it opens up for MITM attacks.

**Comment by csubbiah** [comment]
Thanks @edenhill from openssl perspective "can occur when you encrypt and decrypt with an incompatible versions of openssl" Will check with the issuer which message digest was used for encryption.

Could you also let us know nuget package to use for SchemaRegistry and AVRO serializer, deserializer which uses .net 4.5.2 framework only.

**Comment by csubbiah** [comment]
@edenhill  Thanks. On further analysis we got to know the Java KeyStorePassword and Keypassword (Private Key Password) could be different in a JKS file. However in OpenSSL (PKCS12) which is used by confluent.net API uses underlying openssl - it requires keystorepassword and keypassword to be same. This was  the reason for the above error. Once we updated the password in PKCS12 file we were able to get rid of this error.
